multiple condor jobs on same system interfere with each other

**Describe the bug**
If I run two condor jobs at once on the same system, I get the following error

```
==> runinfo/043/parsl.log <==
2021-12-05 08:59:43.043 parsl.dataflow.flow_control:114 [ERROR]  Flow control callback threw an exception - logging and proceeding anyway
Traceback (most recent call last):
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/dataflow/flow_control.py", line 112, in make_callback
    self.callback(tasks=self._event_buffer, kind=kind)
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/dataflow/task_status_poller.py", line 102, in poll
    self._update_state()
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/dataflow/task_status_poller.py", line 109, in _update_state
    item.poll(now)
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/dataflow/task_status_poller.py", line 44, in poll
    self._status = self._executor.status()
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/executors/status_handling.py", line 73, in status
    status = self._make_status_dict(block_ids, self._provider.status(job_ids))
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/providers/condor/condor.py", line 166, in status
    self._status()
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/providers/condor/condor.py", line 150, in _status
    self.resources[job_id]['status'] = JobStatus(state)
KeyError: '69288.0'

==> runinfo/042/parsl.log <==
2021-12-05 08:59:45.521 parsl.dataflow.strategy:210 [DEBUG]  Executor HighThroughputExecutor has 200 active tasks, 0/200 running/pending blocks, and 0 connected workers
```

It appears that one job is seeing the pilot blocks of the other

**To Reproduce**
Run two condor jobs. You may need to kill one or the other and restart them.

**Expected behavior**
A clear and concise description of what you expected to happen.

**Environment**
 - OS: linux
 - Python version: 3.7.12 
 - parsl version: 1.1.0

**Distributed Environment**
- Where are you running the Parsl script from ? Login node
- Where do you need the workers to run ? Condor

Here's an example with 4 parsl runs all looking for the same condor pilot job:

```
==> runinfo/044/parsl.log <==
2021-12-05 09:29:54.131 parsl.dataflow.flow_control:114 [ERROR]  Flow control callback threw an exception - logging and proceeding anyway
Traceback (most recent call last):
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/dataflow/flow_control.py", line 112, in make_callback
    self.callback(tasks=self._event_buffer, kind=kind)
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/dataflow/task_status_poller.py", line 102, in poll
    self._update_state()
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/dataflow/task_status_poller.py", line 109, in _update_state
    item.poll(now)
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/dataflow/task_status_poller.py", line 44, in poll
    self._status = self._executor.status()
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/executors/status_handling.py", line 73, in status
    status = self._make_status_dict(block_ids, self._provider.status(job_ids))
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/providers/condor/condor.py", line 166, in status
    self._status()
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/providers/condor/condor.py", line 150, in _status
    self.resources[job_id]['status'] = JobStatus(state)
KeyError: '69674.0'

==> runinfo/045/parsl.log <==
2021-12-05 09:29:55.883 parsl.dataflow.flow_control:114 [ERROR]  Flow control callback threw an exception - logging and proceeding anyway
Traceback (most recent call last):
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/dataflow/flow_control.py", line 112, in make_callback
    self.callback(tasks=self._event_buffer, kind=kind)
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/dataflow/task_status_poller.py", line 102, in poll
    self._update_state()
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/dataflow/task_status_poller.py", line 109, in _update_state
    item.poll(now)
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/dataflow/task_status_poller.py", line 44, in poll
    self._status = self._executor.status()
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/executors/status_handling.py", line 73, in status
    status = self._make_status_dict(block_ids, self._provider.status(job_ids))
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/providers/condor/condor.py", line 166, in status
    self._status()
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/providers/condor/condor.py", line 150, in _status
    self.resources[job_id]['status'] = JobStatus(state)
KeyError: '69674.0'

==> runinfo/046/parsl.log <==
2021-12-05 09:29:55.892 parsl.dataflow.flow_control:114 [ERROR]  Flow control callback threw an exception - logging and proceeding anyway
Traceback (most recent call last):
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/dataflow/flow_control.py", line 112, in make_callback
    self.callback(tasks=self._event_buffer, kind=kind)
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/dataflow/task_status_poller.py", line 102, in poll
    self._update_state()
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/dataflow/task_status_poller.py", line 109, in _update_state
    item.poll(now)
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/dataflow/task_status_poller.py", line 44, in poll
    self._status = self._executor.status()
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/executors/status_handling.py", line 73, in status
    status = self._make_status_dict(block_ids, self._provider.status(job_ids))
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/providers/condor/condor.py", line 166, in status
    self._status()
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/providers/condor/condor.py", line 150, in _status
    self.resources[job_id]['status'] = JobStatus(state)
KeyError: '69674.0'

==> runinfo/047/parsl.log <==
2021-12-05 09:29:56.210 parsl.dataflow.flow_control:114 [ERROR]  Flow control callback threw an exception - logging and proceeding anyway
Traceback (most recent call last):
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/dataflow/flow_control.py", line 112, in make_callback
    self.callback(tasks=self._event_buffer, kind=kind)
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/dataflow/task_status_poller.py", line 102, in poll
    self._update_state()
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/dataflow/task_status_poller.py", line 109, in _update_state
    item.poll(now)
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/dataflow/task_status_poller.py", line 44, in poll
    self._status = self._executor.status()
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/executors/status_handling.py", line 73, in status
    status = self._make_status_dict(block_ids, self._provider.status(job_ids))
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/providers/condor/condor.py", line 166, in status
    self._status()
  File "/gpfs02/astro/workarea/beckermr/miniconda3/lib/python3.7/site-packages/parsl/providers/condor/condor.py", line 150, in _status
    self.resources[job_id]['status'] = JobStatus(state)
KeyError: '69674.0'
```

--- a/codemeta.json
+++ b/codemeta.json
@@ -58,7 +58,7 @@
             "givenName": "Ian",
             "familyName": "Foster",
             "email": "foster@anl.gov"
-        }, 
+        },
         {
             "@id":"http://orcid.org/0000-0002-0762-2684",
             "@type": "Person",
@@ -72,6 +72,14 @@
             "givenName": "Kelly L.",
             "familyName": "Rowland",
             "email": "kellyrowland@lbl.gov"
+        },
+        {
+            "@id":"https://orcid.org/0000-0001-7774-2246",
+            "@type": "Person",
+            "givenName": "Matthew R.",
+            "familyName": "Becker",
+            "email": "becker.mr@gmail.com"
+        }
     ],
     "softwareRequirements": [
         {

--- a/parsl/providers/condor/condor.py
+++ b/parsl/providers/condor/condor.py
@@ -74,7 +74,9 @@ class CondorProvider(RepresentationMixin, ClusterProvider):
         Launcher for this provider. Possible launchers include
         :class:`~parsl.launchers.SingleNodeLauncher` (the default),
     cmd_timeout : int
-        Timeout for commands made to the scheduler in seconds
+        Timeout for commands made to the scheduler in seconds.
+    cmd_chunk_size : int
+        Calls to the scheduler will be made for chunks of blocks with this size.
     """
     @typeguard.typechecked
     def __init__(self,
@@ -94,7 +96,8 @@ class CondorProvider(RepresentationMixin, ClusterProvider):
                  worker_init: str = '',
                  launcher: Launcher = SingleNodeLauncher(),
                  requirements: str = '',
-                 cmd_timeout: int = 60) -> None:
+                 cmd_timeout: int = 60,
+                 cmd_chunk_size: int = 100) -> None:
 
         label = 'condor'
         super().__init__(label,
@@ -109,6 +112,7 @@ class CondorProvider(RepresentationMixin, ClusterProvider):
                          cmd_timeout=cmd_timeout)
         self.cores_per_slot = cores_per_slot
         self.mem_per_slot = mem_per_slot
+        self.cmd_chunk_size = cmd_chunk_size
 
         # To Parsl, Condor slots should be treated equivalently to nodes
         self.cores_per_node = cores_per_slot
@@ -132,22 +136,24 @@ class CondorProvider(RepresentationMixin, ClusterProvider):
     def _status(self):
         """Update the resource dictionary with job statuses."""
 
-        job_id_list = ' '.join(self.resources.keys())
-        cmd = "condor_q {0} -af:jr JobStatus".format(job_id_list)
-        retcode, stdout, stderr = self.execute_wait(cmd)
-        """
-        Example output:
+        for job_id_chunk in _chunker(self.resources.keys(), self.cmd_chunk_size):
+            job_id_list = ' '.join(job_id_chunk)
+            cmd = "condor_q {0} -af:jr JobStatus".format(job_id_list)
+            retcode, stdout, stderr = self.execute_wait(cmd)
+            """
+            Example output:
 
-        $ condor_q 34524642.0 34524643.0 -af:jr JobStatus
-        34524642.0 2
-        34524643.0 1
-        """
+            $ condor_q 34524642.0 34524643.0 -af:jr JobStatus
+            34524642.0 2
+            34524643.0 1
+            """
 
-        for line in stdout.splitlines():
-            parts = line.strip().split()
-            job_id = parts[0]
-            state = translate_table.get(parts[1], JobState.UNKNOWN)
-            self.resources[job_id]['status'] = JobStatus(state)
+            for line in stdout.splitlines():
+                parts = line.strip().split()
+                job_id = parts[0]
+                state = translate_table.get(parts[1], JobState.UNKNOWN)
+                if job_id in self.resources:
+                    self.resources[job_id]['status'] = JobStatus(state)
 
     def status(self, job_ids):
         """Get the status of a list of jobs identified by their ids.
@@ -290,18 +296,19 @@ class CondorProvider(RepresentationMixin, ClusterProvider):
         list of bool
             Each entry in the list will be True if the job is cancelled succesfully, otherwise False.
         """
-
-        job_id_list = ' '.join(job_ids)
-        cmd = "condor_rm {0}; condor_rm -forcex {0}".format(job_id_list)
-        logger.debug("Attempting removal of jobs : {0}".format(cmd))
-        retcode, stdout, stderr = self.execute_wait(cmd)
-        rets = None
-        if retcode == 0:
-            for jid in job_ids:
-                self.resources[jid]['status'] = JobStatus(JobState.CANCELLED)
-            rets = [True for i in job_ids]
-        else:
-            rets = [False for i in job_ids]
+        rets = []
+        for job_id_chunk in _chunker(job_ids, self.cmd_chunk_size):
+            job_id_list = ' '.join(job_id_chunk)
+            cmd = "condor_rm {0}; condor_rm -forcex {0}".format(job_id_list)
+            logger.debug("Attempting removal of jobs : {0}".format(cmd))
+            retcode, stdout, stderr = self.execute_wait(cmd)
+            if retcode == 0:
+                for jid in job_ids:
+                    if jid in self.resources:
+                        self.resources[jid]['status'] = JobStatus(JobState.CANCELLED)
+                rets.extend([True for i in job_ids])
+            else:
+                rets.extend([False for i in job_ids])
 
         return rets
 
@@ -313,6 +320,18 @@ class CondorProvider(RepresentationMixin, ClusterProvider):
     @property
     def status_polling_interval(self):
         return 60
+
+
+# see https://stackoverflow.com/a/18243990
+def _chunker(seq, size):
+    res = []
+    for el in seq:
+        res.append(el)
+        if len(res) == size:
+            yield res
+            res = []
+    if res:
+        yield res
 
 
 if __name__ == "__main__":

--- a/parsl/providers/condor/condor.py
+++ b/parsl/providers/condor/condor.py
@@ -303,12 +303,12 @@ class CondorProvider(RepresentationMixin, ClusterProvider):
             logger.debug("Attempting removal of jobs : {0}".format(cmd))
             retcode, stdout, stderr = self.execute_wait(cmd)
             if retcode == 0:
-                for jid in job_ids:
+                for jid in job_id_chunk:
                     if jid in self.resources:
                         self.resources[jid]['status'] = JobStatus(JobState.CANCELLED)
-                rets.extend([True for i in job_ids])
+                rets.extend([True for i in job_id_chunk])
             else:
-                rets.extend([False for i in job_ids])
+                rets.extend([False for i in job_id_chunk])
 
         return rets
 

